apps/ledger-live-mobile/src/newArch/features/Market/screens/MarketDetail/MarketDetailCont.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useSingleCoinMarketData } from "@ledgerhq/live-common/market/MarketDataProvider";
+import { readOnlyModeEnabledSelector, starredMarketCoinsSelector } from "~/reducers/settings";
+import { addStarredMarketCoins, removeStarredMarketCoins } from "~/actions/settings";
+import { flattenAccountsByCryptoCurrencyScreenSelector } from "~/reducers/accounts";
+import { screen, track } from "~/analytics";
+import { ScreenName } from "~/const";
+import useNotifications from "~/logic/notifications";
+import { BaseComposite, StackNavigatorProps } from "~/components/RootNavigator/types/helpers";
+import { MarketNavigatorStackParamList } from "LLM/features/Market/Navigator";
+import MarketDetails from "./MarketDetail";
+
+type NavigationProps = BaseComposite<
+  StackNavigatorProps<MarketNavigatorStackParamList, ScreenName.MarketDetail>
+>;
+
+function MarketDetailCont({ navigation, route }: NavigationProps) {
+  const { params } = route;
+  const { currencyId, resetSearchOnUmount } = params;
+  const dispatch = useDispatch();
+  const starredMarketCoins: string[] = useSelector(starredMarketCoinsSelector);
+  const isStarred = starredMarketCoins.includes(currencyId);
+  const { triggerMarketPushNotificationModal } = useNotifications();
+  const [hasRetried, setHasRetried] = useState<boolean>(false);
+
+  const {
+    selectedCoinData: currency,
+    selectCurrency,
+    chartRequestParams,
+    loading,
+    loadingChart,
+    refreshChart,
+    counterCurrency,
+  } = useSingleCoinMarketData();
+
+  const { name, internalCurrency } = currency || {};
+
+  useEffect(() => {
+    if (!loading) {
+      if (currency === undefined && !hasRetried) {
+        selectCurrency(currencyId);
+        setHasRetried(true);
+      } else if (currency && hasRetried) {
+        setHasRetried(false);
+      }
+    }
+  }, [currency, selectCurrency, currencyId, hasRetried, loading]);
+
+  useEffect(() => {
+    const resetState = () => {
+      // selectCurrency();
+    };
+    const sub = navigation.addListener("blur", resetState);
+    return () => {
+      sub();
+    };
+  }, [selectCurrency, resetSearchOnUmount, navigation]);
+
+  const allAccounts = useSelector(flattenAccountsByCryptoCurrencyScreenSelector(internalCurrency));
```

Comment:
do you know if allAccounts ref will change on each render() or if it's cached and only recalc if accounts store changes? i suspect it may cascade to perf issue if it's not the case

Reply:
We do have an issue here: we recompute it 11x when we navigate to the details page. I will investigate

Reply:
The issue we have is that we create a new object each time we call `flattenAccountsByCryptoCurrencyScreenSelector` => `{currencies: [currency.id]}`
So it breaks the createSelector cache mechanism. I fixed it by only using currency.id as a parameter.

```
export const flattenAccountsByCryptoCurrencyScreenSelector =
  (currency?: CryptoCurrency | TokenCurrency) => (state: State) => {
    if (!currency) return emptyArray;
    return flattenAccountsByCryptoCurrencySelector(state, {
      currencies: [currency.id],
    });
  };
  ```
  
I think that we have this kind of issue in a lot of other places.

Reply:
Ok, so I fixed this issue by fixing our reselect usage.
Today we recreate a new object, or a new array for parameters we send to createSelector.
It's documented here on how to deal with multiple arguments: https://reselect.js.org/faq/#how-do-i-create-a-selector-that-takes-an-argument

I was not able to fix all selectors as I have this TS error https://github.com/reduxjs/reselect/issues/550.
Can be the subject of a dedicated improvement.